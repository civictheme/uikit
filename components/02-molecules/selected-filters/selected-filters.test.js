const template = 'components/02-molecules/selected-filters/selected-filters.twig';

describe('Selected Filters Component', () => {
  test('renders with required attributes', async () => {
    const c = await dom(template, {
      title: 'Selected filters:',
      filters: [
        {
          text: 'Category: News',
          url: '#',
          label: 'Remove filter: Category News',
        },
        {
          text: 'Topic: Technology',
          url: '#',
          label: 'Remove filter: Topic Technology',
        },
      ],
    });

    expect(c.querySelectorAll('.ct-selected-filters')).toHaveLength(1);
    expect(c.querySelector('.ct-selected-filters__title').textContent.trim()).toEqual('Selected filters:');

    const items = c.querySelectorAll('.ct-selected-filters__list .ct-chip');
    expect(items).toHaveLength(2);
    expect(items[0].textContent.trim()).toContain('Category: News');
    expect(items[0].getAttribute('href')).toEqual('#');
    expect(items[0].getAttribute('alt')).toEqual('Remove filter: Category News');
    expect(items[1].textContent.trim()).toContain('Topic: Technology');
    expect(items[1].getAttribute('href')).toEqual('#');
    expect(items[1].getAttribute('alt')).toEqual('Remove filter: Topic Technology');

    assertUniqueCssClasses(c);
  });

  test('renders with clear all link', async () => {
    const c = await dom(template, {
      title: 'Selected filters:',
      filters: [
        {
          text: 'Category: News',
          url: '#',
          label: 'Remove filter: Category News',
        },
      ],
      clear_link: {
        text: 'Clear all',
        url: '#clear',
        type: 'secondary',
      },
    });

    const clearButton = c.querySelector('.ct-selected-filters__clear .ct-button');
    expect(clearButton).toBeTruthy();
    expect(clearButton.textContent.trim()).toContain('Clear all');
    expect(clearButton.getAttribute('href')).toEqual('#clear');
    expect(clearButton.classList.contains('ct-button--secondary')).toBe(true);
    expect(clearButton.classList.contains('ct-button--regular')).toBe(true);

    assertUniqueCssClasses(c);
  });

  test('renders with dark theme', async () => {
    const c = await dom(template, {
      theme: 'dark',
      title: 'Selected filters:',
      filters: [
        {
          text: 'Category: News',
          url: '#',
          label: 'Remove filter: Category News',
        },
      ],
    });

    expect(c.querySelector('.ct-selected-filters').classList.contains('ct-theme-dark')).toBe(true);
    expect(c.querySelector('.ct-chip').classList.contains('ct-theme-dark')).toBe(true);

    assertUniqueCssClasses(c);
  });

  test('renders empty when no filters provided', async () => {
    const c = await dom(template, {
      title: 'Selected filters:',
      filters: [],
    });

    expect(c.querySelector('.ct-selected-filters__list .ct-chip')).toBeFalsy();
    expect(c.querySelector('.ct-selected-filters__clear')).toBeFalsy();

    assertUniqueCssClasses(c);
  });
});
