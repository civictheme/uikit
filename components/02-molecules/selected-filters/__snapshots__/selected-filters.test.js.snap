// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Selected Filters Component renders empty when no filters provided 1`] = `
<div>
  



  <div
    class="ct-selected-filters ct-theme-light "
  >
    
  
    <div
      class="ct-selected-filters__active-filters"
    >
      
    
      <div
        class="ct-selected-filters__title"
        id="selected-filters"
      >
        Selected filters:
      </div>
      
            

  
    </div>
    
  
  </div>
  

</div>
`;

exports[`Selected Filters Component renders with clear all link 1`] = `
<div>
  



  <div
    class="ct-selected-filters ct-theme-light "
  >
    
  
    <div
      class="ct-selected-filters__active-filters"
    >
      
    
      <div
        class="ct-selected-filters__title"
        id="selected-filters"
      >
        Selected filters:
      </div>
      
                            

  
      <ul
        aria-labelledby="selected-filters"
        class="ct-item-list ct-item-list--horizontal ct-item-list--small  ct-selected-filters__list"
      >
        
                  
        <li
          class="ct-item-list__item"
        >
                  
      
          <a
            alt="Remove filter: Category News"
            class="ct-chip ct-theme-light ct-chip--link ct-chip--small active  "
            data-component-name="chip"
            href="#"
          >
            Category: News        
            <svg
              aria-hidden="true"
              class="ct-icon  ct-chip__dismiss"
              height="24"
              role="img"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              

              <path
                d="M13.4099 11.9999L17.7099 7.70994C17.8982 7.52164 18.004 7.26624 18.004 6.99994C18.004 6.73364 17.8982 6.47825 17.7099 6.28994C17.5216 6.10164 17.2662 5.99585 16.9999 5.99585C16.7336 5.99585 16.4782 6.10164 16.2899 6.28994L11.9999 10.5899L7.70994 6.28994C7.52164 6.10164 7.26624 5.99585 6.99994 5.99585C6.73364 5.99585 6.47824 6.10164 6.28994 6.28994C6.10164 6.47825 5.99585 6.73364 5.99585 6.99994C5.99585 7.26624 6.10164 7.52164 6.28994 7.70994L10.5899 11.9999L6.28994 16.2899C6.19621 16.3829 6.12182 16.4935 6.07105 16.6154C6.02028 16.7372 5.99414 16.8679 5.99414 16.9999C5.99414 17.132 6.02028 17.2627 6.07105 17.3845C6.12182 17.5064 6.19621 17.617 6.28994 17.7099C6.3829 17.8037 6.4935 17.8781 6.61536 17.9288C6.73722 17.9796 6.86793 18.0057 6.99994 18.0057C7.13195 18.0057 7.26266 17.9796 7.38452 17.9288C7.50638 17.8781 7.61698 17.8037 7.70994 17.7099L11.9999 13.4099L16.2899 17.7099C16.3829 17.8037 16.4935 17.8781 16.6154 17.9288C16.7372 17.9796 16.8679 18.0057 16.9999 18.0057C17.132 18.0057 17.2627 17.9796 17.3845 17.9288C17.5064 17.8781 17.617 17.8037 17.7099 17.7099C17.8037 17.617 17.8781 17.5064 17.9288 17.3845C17.9796 17.2627 18.0057 17.132 18.0057 16.9999C18.0057 16.8679 17.9796 16.7372 17.9288 16.6154C17.8781 16.4935 17.8037 16.3829 17.7099 16.2899L13.4099 11.9999Z"
              />
              

            </svg>
            
          
          </a>
          
        
        </li>
        
            
      </ul>
      
  
    </div>
    
      
    <div
      class="ct-selected-filters__clear"
    >
      
      

  
      <a
        class="ct-button ct-theme-light ct-button--secondary ct-button--link ct-button--small   "
        data-component-name="button"
        href="#clear"
        role="button"
      >
        Clear 
        <span
          class="ct-text-icon__group"
        >
          all 
          <svg
            aria-hidden="true"
            class="ct-icon  ct-button__icon"
            height="24"
            role="img"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            

            <path
              d="M15.7098 8.28996C15.6169 8.19623 15.5063 8.12183 15.3844 8.07107C15.2626 8.0203 15.1318 7.99416 14.9998 7.99416C14.8678 7.99416 14.7371 8.0203 14.6153 8.07107C14.4934 8.12183 14.3828 8.19623 14.2898 8.28996L11.9998 10.59L9.70984 8.28996C9.52153 8.10165 9.26614 7.99587 8.99984 7.99587C8.73353 7.99587 8.47814 8.10165 8.28983 8.28996C8.10153 8.47826 7.99574 8.73366 7.99574 8.99996C7.99574 9.26626 8.10153 9.52165 8.28983 9.70996L10.5898 12L8.28983 14.29C8.19611 14.3829 8.12171 14.4935 8.07094 14.6154C8.02017 14.7372 7.99404 14.8679 7.99404 15C7.99404 15.132 8.02017 15.2627 8.07094 15.3845C8.12171 15.5064 8.19611 15.617 8.28983 15.71C8.3828 15.8037 8.4934 15.8781 8.61526 15.9288C8.73712 15.9796 8.86782 16.0058 8.99984 16.0058C9.13185 16.0058 9.26255 15.9796 9.38441 15.9288C9.50627 15.8781 9.61687 15.8037 9.70984 15.71L11.9998 13.41L14.2898 15.71C14.3828 15.8037 14.4934 15.8781 14.6153 15.9288C14.7371 15.9796 14.8678 16.0058 14.9998 16.0058C15.1318 16.0058 15.2626 15.9796 15.3844 15.9288C15.5063 15.8781 15.6169 15.8037 15.7098 15.71C15.8036 15.617 15.878 15.5064 15.9287 15.3845C15.9795 15.2627 16.0056 15.132 16.0056 15C16.0056 14.8679 15.9795 14.7372 15.9287 14.6154C15.878 14.4935 15.8036 14.3829 15.7098 14.29L13.4098 12L15.7098 9.70996C15.8036 9.61699 15.878 9.50639 15.9287 9.38453C15.9795 9.26267 16.0056 9.13197 16.0056 8.99996C16.0056 8.86795 15.9795 8.73724 15.9287 8.61538C15.878 8.49352 15.8036 8.38292 15.7098 8.28996ZM19.0698 4.92996C18.1474 3.97486 17.0439 3.21303 15.8239 2.68894C14.6038 2.16485 13.2916 1.88899 11.9638 1.87745C10.6361 1.86591 9.31926 2.11893 8.09029 2.62174C6.86133 3.12455 5.74481 3.86708 4.80589 4.80601C3.86696 5.74493 3.12443 6.86145 2.62162 8.09042C2.11881 9.31938 1.86579 10.6362 1.87733 11.964C1.88887 13.2918 2.16473 14.604 2.68882 15.824C3.21291 17.044 3.97473 18.1475 4.92984 19.07C5.8523 20.0251 6.95575 20.7869 8.17579 21.311C9.39583 21.8351 10.708 22.1109 12.0358 22.1225C13.3636 22.134 14.6804 21.881 15.9094 21.3782C17.1383 20.8754 18.2549 20.1328 19.1938 19.1939C20.1327 18.255 20.8752 17.1385 21.3781 15.9095C21.8809 14.6805 22.1339 13.3637 22.1223 12.0359C22.1108 10.7082 21.8349 9.39595 21.3109 8.17591C20.7868 6.95587 20.0249 5.85243 19.0698 4.92996ZM17.6598 17.66C16.3519 18.9694 14.6304 19.7848 12.7886 19.9673C10.9469 20.1497 9.09884 19.688 7.55936 18.6607C6.01987 17.6334 4.88419 16.1042 4.34581 14.3334C3.80742 12.5627 3.89964 10.6601 4.60675 8.94974C5.31386 7.23938 6.59211 5.82711 8.22373 4.95352C9.85534 4.07993 11.7394 3.79909 13.5548 4.15883C15.3703 4.51857 17.0049 5.49665 18.1801 6.92642C19.3553 8.35619 19.9984 10.1492 19.9998 12C20.0034 13.0512 19.7984 14.0928 19.3968 15.0643C18.9951 16.0359 18.4047 16.9181 17.6598 17.66Z"
            />
            

          </svg>
          

        </span>
      </a>
      
    
    </div>
    
  
  </div>
  

</div>
`;

exports[`Selected Filters Component renders with dark theme 1`] = `
<div>
  



  <div
    class="ct-selected-filters ct-theme-dark "
  >
    
  
    <div
      class="ct-selected-filters__active-filters"
    >
      
    
      <div
        class="ct-selected-filters__title"
        id="selected-filters"
      >
        Selected filters:
      </div>
      
                            

  
      <ul
        aria-labelledby="selected-filters"
        class="ct-item-list ct-item-list--horizontal ct-item-list--small  ct-selected-filters__list"
      >
        
                  
        <li
          class="ct-item-list__item"
        >
                  
      
          <a
            alt="Remove filter: Category News"
            class="ct-chip ct-theme-dark ct-chip--link ct-chip--small active  "
            data-component-name="chip"
            href="#"
          >
            Category: News        
            <svg
              aria-hidden="true"
              class="ct-icon  ct-chip__dismiss"
              height="24"
              role="img"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              

              <path
                d="M13.4099 11.9999L17.7099 7.70994C17.8982 7.52164 18.004 7.26624 18.004 6.99994C18.004 6.73364 17.8982 6.47825 17.7099 6.28994C17.5216 6.10164 17.2662 5.99585 16.9999 5.99585C16.7336 5.99585 16.4782 6.10164 16.2899 6.28994L11.9999 10.5899L7.70994 6.28994C7.52164 6.10164 7.26624 5.99585 6.99994 5.99585C6.73364 5.99585 6.47824 6.10164 6.28994 6.28994C6.10164 6.47825 5.99585 6.73364 5.99585 6.99994C5.99585 7.26624 6.10164 7.52164 6.28994 7.70994L10.5899 11.9999L6.28994 16.2899C6.19621 16.3829 6.12182 16.4935 6.07105 16.6154C6.02028 16.7372 5.99414 16.8679 5.99414 16.9999C5.99414 17.132 6.02028 17.2627 6.07105 17.3845C6.12182 17.5064 6.19621 17.617 6.28994 17.7099C6.3829 17.8037 6.4935 17.8781 6.61536 17.9288C6.73722 17.9796 6.86793 18.0057 6.99994 18.0057C7.13195 18.0057 7.26266 17.9796 7.38452 17.9288C7.50638 17.8781 7.61698 17.8037 7.70994 17.7099L11.9999 13.4099L16.2899 17.7099C16.3829 17.8037 16.4935 17.8781 16.6154 17.9288C16.7372 17.9796 16.8679 18.0057 16.9999 18.0057C17.132 18.0057 17.2627 17.9796 17.3845 17.9288C17.5064 17.8781 17.617 17.8037 17.7099 17.7099C17.8037 17.617 17.8781 17.5064 17.9288 17.3845C17.9796 17.2627 18.0057 17.132 18.0057 16.9999C18.0057 16.8679 17.9796 16.7372 17.9288 16.6154C17.8781 16.4935 17.8037 16.3829 17.7099 16.2899L13.4099 11.9999Z"
              />
              

            </svg>
            
          
          </a>
          
        
        </li>
        
            
      </ul>
      
  
    </div>
    
  
  </div>
  

</div>
`;

exports[`Selected Filters Component renders with required attributes 1`] = `
<div>
  



  <div
    class="ct-selected-filters ct-theme-light "
  >
    
  
    <div
      class="ct-selected-filters__active-filters"
    >
      
    
      <div
        class="ct-selected-filters__title"
        id="selected-filters"
      >
        Selected filters:
      </div>
      
                                            

  
      <ul
        aria-labelledby="selected-filters"
        class="ct-item-list ct-item-list--horizontal ct-item-list--small  ct-selected-filters__list"
      >
        
                  
        <li
          class="ct-item-list__item"
        >
                  
      
          <a
            alt="Remove filter: Category News"
            class="ct-chip ct-theme-light ct-chip--link ct-chip--small active  "
            data-component-name="chip"
            href="#"
          >
            Category: News        
            <svg
              aria-hidden="true"
              class="ct-icon  ct-chip__dismiss"
              height="24"
              role="img"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              

              <path
                d="M13.4099 11.9999L17.7099 7.70994C17.8982 7.52164 18.004 7.26624 18.004 6.99994C18.004 6.73364 17.8982 6.47825 17.7099 6.28994C17.5216 6.10164 17.2662 5.99585 16.9999 5.99585C16.7336 5.99585 16.4782 6.10164 16.2899 6.28994L11.9999 10.5899L7.70994 6.28994C7.52164 6.10164 7.26624 5.99585 6.99994 5.99585C6.73364 5.99585 6.47824 6.10164 6.28994 6.28994C6.10164 6.47825 5.99585 6.73364 5.99585 6.99994C5.99585 7.26624 6.10164 7.52164 6.28994 7.70994L10.5899 11.9999L6.28994 16.2899C6.19621 16.3829 6.12182 16.4935 6.07105 16.6154C6.02028 16.7372 5.99414 16.8679 5.99414 16.9999C5.99414 17.132 6.02028 17.2627 6.07105 17.3845C6.12182 17.5064 6.19621 17.617 6.28994 17.7099C6.3829 17.8037 6.4935 17.8781 6.61536 17.9288C6.73722 17.9796 6.86793 18.0057 6.99994 18.0057C7.13195 18.0057 7.26266 17.9796 7.38452 17.9288C7.50638 17.8781 7.61698 17.8037 7.70994 17.7099L11.9999 13.4099L16.2899 17.7099C16.3829 17.8037 16.4935 17.8781 16.6154 17.9288C16.7372 17.9796 16.8679 18.0057 16.9999 18.0057C17.132 18.0057 17.2627 17.9796 17.3845 17.9288C17.5064 17.8781 17.617 17.8037 17.7099 17.7099C17.8037 17.617 17.8781 17.5064 17.9288 17.3845C17.9796 17.2627 18.0057 17.132 18.0057 16.9999C18.0057 16.8679 17.9796 16.7372 17.9288 16.6154C17.8781 16.4935 17.8037 16.3829 17.7099 16.2899L13.4099 11.9999Z"
              />
              

            </svg>
            
          
          </a>
          
        
        </li>
        
                        
        <li
          class="ct-item-list__item"
        >
                  
      
          <a
            alt="Remove filter: Topic Technology"
            class="ct-chip ct-theme-light ct-chip--link ct-chip--small active  "
            data-component-name="chip"
            href="#"
          >
            Topic: Technology        
            <svg
              aria-hidden="true"
              class="ct-icon  ct-chip__dismiss"
              height="24"
              role="img"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              

              <path
                d="M13.4099 11.9999L17.7099 7.70994C17.8982 7.52164 18.004 7.26624 18.004 6.99994C18.004 6.73364 17.8982 6.47825 17.7099 6.28994C17.5216 6.10164 17.2662 5.99585 16.9999 5.99585C16.7336 5.99585 16.4782 6.10164 16.2899 6.28994L11.9999 10.5899L7.70994 6.28994C7.52164 6.10164 7.26624 5.99585 6.99994 5.99585C6.73364 5.99585 6.47824 6.10164 6.28994 6.28994C6.10164 6.47825 5.99585 6.73364 5.99585 6.99994C5.99585 7.26624 6.10164 7.52164 6.28994 7.70994L10.5899 11.9999L6.28994 16.2899C6.19621 16.3829 6.12182 16.4935 6.07105 16.6154C6.02028 16.7372 5.99414 16.8679 5.99414 16.9999C5.99414 17.132 6.02028 17.2627 6.07105 17.3845C6.12182 17.5064 6.19621 17.617 6.28994 17.7099C6.3829 17.8037 6.4935 17.8781 6.61536 17.9288C6.73722 17.9796 6.86793 18.0057 6.99994 18.0057C7.13195 18.0057 7.26266 17.9796 7.38452 17.9288C7.50638 17.8781 7.61698 17.8037 7.70994 17.7099L11.9999 13.4099L16.2899 17.7099C16.3829 17.8037 16.4935 17.8781 16.6154 17.9288C16.7372 17.9796 16.8679 18.0057 16.9999 18.0057C17.132 18.0057 17.2627 17.9796 17.3845 17.9288C17.5064 17.8781 17.617 17.8037 17.7099 17.7099C17.8037 17.617 17.8781 17.5064 17.9288 17.3845C17.9796 17.2627 18.0057 17.132 18.0057 16.9999C18.0057 16.8679 17.9796 16.7372 17.9288 16.6154C17.8781 16.4935 17.8037 16.3829 17.7099 16.2899L13.4099 11.9999Z"
              />
              

            </svg>
            
          
          </a>
          
        
        </li>
        
            
      </ul>
      
  
    </div>
    
  
  </div>
  

</div>
`;
